File: awa/sdk_dist/python/awa/client/utils/activity/delete_directory_activity.py
from datetime import timedelta
from pathlib import Path

from temporalio import workflow

from awa.client import constants


async def delete_directory_activity(dir_path: str | Path) -> None:
    """Delete a directory and all its contents.

    Args:
        dir_path: The directory path to delete.

    """
    await workflow.execute_activity(
        constants.ACTIVITY_DELETE_DIRECTORY,
        args=[str(dir_path)],
        task_queue=constants.AWA_DEFAULT_TASK_QUEUE,
        start_to_close_timeout=timedelta(seconds=constants.FILE_IO_TIMEOUT_SECONDS * 10),
    )

File: awa/sdk_dist/csharp/Awa.Client/Utilities/Activity/DeleteDirectoryActivity.cs
using System;
using System.Threading.Tasks;
using Awa.Client.Constants;

namespace Awa.Client.Utilities.Activity
{
    public static partial class ActivityUtilities
    {
        /// <summary>
        /// Delete a directory and all its contents.
        /// </summary>
        /// <param name="dirPath">The directory path to delete.</param>
        public static async Task DeleteDirectoryActivity(string dirPath)
        {
            await Temporalio.Workflows.Workflow.ExecuteActivityAsync(
                AwaConstants.ActivityDeleteDirectory,
                new object?[] { dirPath },
                new()
                {
                    TaskQueue = AwaConstants.AwaDefaultTaskQueue,
                    StartToCloseTimeout = TimeSpan.FromSeconds(AwaConstants.FileIoTimeoutSeconds * 10)
                });
        }
    }
}
